src/test/java/com/google/neighborgood/UserInfoServletTest.java

```diff
@@ -0,0 +1,422 @@
+    assertEquals("leonardzhang@google.com", (String) entity.getProperty("email"));
+    assertEquals("1234567890", (String) entity.getProperty("userId"));
+    assertEquals(0, (long) entity.getProperty("points"));
+  }
+
+  @Test
+  public void doGetTest() throws IOException {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    // Check whether the datastore is empty before the test
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    assertEquals(0, ds.prepare(new Query("UserInfo")).countEntities(withLimit(10)));
+
+    Entity dummy = new Entity("UserInfo");
+    dummy.setProperty("nickname", "Leonard");
+    dummy.setProperty("address", "xxx");
+    dummy.setProperty("phone", "xxx");
+    dummy.setProperty("email", "test@example.com");
+    dummy.setProperty("userId", "1234567890");
+    dummy.setProperty("points", 0);
+    ds.put(dummy);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    try {
+      new UserInfoServlet().doGet(request, response);
+    } catch (IOException e) {
+      assertTrue(false);
+    }
+
+    // After sending the GET request, the doGet function should output the json string
+    writer.flush();
+    assertTrue(stringWriter.toString().contains("[\"Leonard\",\"xxx\",\"xxx\"]"));
+  }
+
+  @Test
+  public void stressTest() {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+    // Insert 999 dummy entities into datastore to simulate 999 users
+    for (int i = 0; i < 999; i++) {
+      String str = Integer.toString(i);
+      Entity dummy = new Entity("UserInfo");
+      dummy.setProperty("nickname", "Leonard" + str);
+      dummy.setProperty("address", "xxx" + str);
+      dummy.setProperty("phone", str + str);
+      dummy.setProperty("email", "test" + str + "@example.com");
+      dummy.setProperty("userId", str);
+      dummy.setProperty("points", 0);
+      ds.put(dummy);
+    }
+
+    // Check the number of entities in the datastore. The result should be 999
+    assertEquals(999, ds.prepare(new Query("UserInfo")).countEntities());
+
+    // Now we insert a target UserInfo Entity and check the time needed to find the entity
+    when(request.getParameter("nickname-input")).thenReturn("Leonard");
+    when(request.getParameter("address-input")).thenReturn("4xxx Centre Avenue");
+    when(request.getParameter("phone-input")).thenReturn("4xxxxxxxxx");
+
+    try {
+      new UserInfoServlet().doPost(request, response);
+    } catch (IOException e) {
+      assertTrue(false);
+    }
+
+    // After sending the POST request, there should be three entities in the datastore
```

Comment:
not three, but one thousand

Reply:
Ah I forgot to change the comments. The stress test is removed in the new commit so the problem has been fixed automatically.